### PR TITLE
WIP - Overhaul

### PR DESCRIPTION
**Pull request by mvdkleijn**

### CODE REVIEW

File: recipes/agent.rb [hunk 1]
```diff
@@ -119,7 +145,7 @@
   action [:enable, :start]
 end
 
-# Setup monit
+Setup monit
```
Comment (mvdkleijn) [inline recipes/agent.rb]: is a comment

File: recipes/apache2.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # Cookbook Name:: bamboo
 # Recipe:: apache2
 #
-# Copyright 2014
+# Copyright 2013, Brian Flad
```
Comment (mvdkleijn) [inline recipes/apache2.rb]: bad copy-paste

File: templates/default/bamboo-agent.init.erb [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# JIRA startup script - adapted from: http://confluence.atlassian.com/display/JIRA/Starting+JIRA+Automatically+on+Linux
```
Comment (mvdkleijn) [inline templates/default/bamboo-agent.init.erb]: bamboo startup script

File: templates/default/web_app.conf.erb [hunk 4]
```diff
@@ -2,71 +2,49 @@
...
-	<% virtual_host_aliases.each do |virtual_host_alias| -%>
+	<% unless bamboo_virtual_host_alias.empty? -%>
+	<% Array(bamboo_virtual_host_alias).each do |virtual_host_alias| -%>
 	ServerAlias <%= virtual_host_alias %>
 	<% end -%>
 	<% end -%>
 	DocumentRoot <%= node['bamboo']['home_dir'] %>
-	
-	<% unless node['bamboo']['apache2']['error_docs']['e503'].empty? -%>
-	ErrorDocument 503 <%= node['bamboo']['apache2']['error_docs']['e503'] %>
```
Comment (mvdkleijn) [inline templates/default/web_app.conf.erb]: Maybe people are using this

File: templates/default/web_app.conf.erb [hunk 5]
```diff
@@ -2,71 +2,49 @@
...
 
-	<% unless node['bamboo']['apache2']['error_docs']['e503'].empty? -%>
-	ErrorDocument 503 <%= node['bamboo']['apache2']['error_docs']['e503'] %>
-	<% end -%>
-
 	CustomLog <%= node['bamboo']['apache2']['ssl']['access_log'].empty? ? node['apache']['log_dir']+"/bamboo-ssl-access.log" : node['bamboo']['apache2']['ssl']['access_log'] %> combined
 	ErrorLog <%= node['bamboo']['apache2']['ssl']['error_log'].empty? ? node['apache']['log_dir']+"/bamboo-ssl-error.log" : node['bamboo']['apache2']['ssl']['error_log'] %>
 	LogLevel warn
-    ProxyRequests Off
-    ProxyPreserveHost On
```
Comment (mvdkleijn) [inline templates/default/web_app.conf.erb]: I think we still need these two lines

File: .rubocop.yml [hunk 6]
```diff
@@ -23,11 +23,11 @@ Style/SpaceBeforeFirstArg:
 #
 # couldn't figure it out so set this
 #
-ClassAndModuleChildren:
-  EnforcedStyle: compact
+#ClassAndModuleChildren:
+#  EnforcedStyle: compact
```
Comment (mvdkleijn) [inline .rubocop.yml]: we can drop this

File: attributes/default.rb [hunk 7]
```diff
@@ -16,47 +16,36 @@
...
 case node[:bamboo][:database][:type]
 when 'mysql'
-  default[:bamboo][:database][:version]           = '5.6'
   default[:bamboo][:database][:host]              = '127.0.0.1'
-  default[:bamboo][:database][:port]              = 3306
-  default[:bamboo][:database][:root_user_name]    = 'root'
   default[:bamboo][:database_type]                = 'MySQL'
+  # default[:bamboo][:database][:version]         = 5.6 # autoselected via helper functions
 when 'postgresql'
   default[:postgresql][:version]                  = '9.4'
```
Comment (mvdkleijn) [inline attributes/default.rb]: for consistency: also move to helper similar to mysql

File: recipes/crowd_sso.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# Update config to activate Crowd's authenticator to enable SSO
+# See: https://confluence.atlassian.com/crowd/integrating-crowd-with-atlassian-bamboo-198785.html
+default_fragment = '<authenticator class="com.atlassian.bamboo.user.authentication.BambooAuthenticator"/>'
```
Comment (mvdkleijn) [inline recipes/crowd_sso.rb]: missing crowd.properties (seraph is done)

File: templates/default/bamboo-agent.sh.erb [hunk 9]
```diff
@@ -1,557 +0,0 @@
...
-
-# If specified, the Wrapper will be run as the specified user.
-# IMPORTANT - Make sure that the user has the required privileges to write
-#  the PID file and wrapper.log files.  Failure to be able to write the log
-#  file will cause the Wrapper to exit without any way to write out an error
-#  message.
-# NOTE - This will set the user which is used to run the Wrapper as well as
-#  the JVM and is not useful in situations where a privileged resource or
-#  port needs to be allocated prior to the user being changed.
-RUN_AS_USER=<%= node[:bamboo][:agent][:user] %>
```
Comment (ojawel) [inline templates/default/bamboo-agent.sh.erb]: As checked default this is uncommented

**Comment by mvdkleijn** [comment]
Add some stuff to the readme to describe the attribute changes / behavioural changes.

**Pull request merged**